### ISSUE TITLE
CRS.to_proj4() not supproting proj6?

### ISSUE THREAD

**Issue by sscherfke** [pyproj4/pyproj#270]
It looks like [CSR.to_pyproj4()](https://github.com/pyproj4/pyproj/blob/master/pyproj/_crs.pyx#L404) is broken with or not ready for Proj 6?

The following code used to run with pyproj 1.9.6 but breaks with 2.1.3:
```python
>>> from pyproj import Proj
>>> Proj({'a': 6371229.0, 'b': 6371229.0, 'lon_0': -10.0, 'o_lat_p': 30.0, 'o_lon_p': 0.0, 'o_proj': 'longlat', 'proj'
: 'ob_tran'})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../lib/python3.7/site-packages/pyproj/proj.py", line 303, in __init__
    cstrencode(self.crs.to_proj4().replace("+type=crs", "").strip())
AttributeError: 'NoneType' object has no attribute 'replace'
```

**Comment by snowman2** [comment]
This is a bug. Thanks for the report!

**Comment by snowman2** [comment]
I believe the issue is related to the order of the `proj` and `o_proj` parameters. Until this is updated, you need to add `proj` first and `o_proj` second in your dict. See: https://github.com/pyproj4/pyproj/blob/aa161c4e9375e727e8bfb7d11362669b4308b5fb/test/test_crs_cf.py#L144

**Comment by snowman2** [comment]
I checked using projinfo:
```
projinfo '+proj=ob_tran +a=6371229.0 +b=6371229.0 +lon_0=-10.0 +o_lat_p=30.0 +o_lon_p=0.0 +o_proj=longlat +type=crs'
PROJ.4 string:
Error when exporting to PROJ string: DerivedGeographicCRS cannot be exported to PROJ string
```
But this works:
```
$ projinfo '+proj=ob_tran +a=6371229.0 +b=6371229.0 +lon_0=-10.0 +o_lat_p=30.0 +o_lon_p=0.0 +o_proj=longlat'
PROJ string:
+proj=ob_tran +a=6371229.0 +b=6371229.0 +lon_0=-10.0 +o_lat_p=30.0 +o_lon_p=0.0 +o_proj=longlat
```

**Comment by snowman2** [comment]
Fix in #272

**Comment by snowman2** [comment]
I am going to close this as the solution is merged into master. Thanks for raising the issue!
